packages/jaeger-ui/src/components/TracePage/TraceStatistics/index.tsx

```diff
@@ -392,79 +188,180 @@ export default class TraceStatistics extends Component<Props, State> {
     return allTableSpansChange;
   };
 
-  renderTableData() {
-    return this.state.tableValue.map(oneSpan => {
-      const {
-        count,
-        total,
-        avg,
-        min,
-        max,
-        selfTotal,
-        selfAvg,
-        selfMin,
-        selfMax,
-        percent,
-        color,
-        searchColor,
-        colorToPercent,
-      } = oneSpan;
-      const values: any[] = [count, total, avg, min, max, selfTotal, selfAvg, selfMin, selfMax, percent];
-      const uid = _.uniqueId('id');
-      if (!oneSpan.isDetail) {
-        return (
-          <MainTableData
-            key={uid}
-            type={oneSpan.type}
-            name={oneSpan.name}
-            searchColor={searchColor}
-            values={values}
-            columnsArray={columnsArray}
-            togglePopup={this.togglePopup}
-            valueNameSelector1={this.state.valueNameSelector1}
-            valueNameSelector2={this.state.valueNameSelector2}
-            color={color}
-            clickColumn={this.clickColumn}
-            colorToPercent={colorToPercent}
-          />
-        );
+  render() {
+    const onClickOption = (type: string, name: string) => {
+      if (this.state.valueNameSelector1 === 'sql.query' && type !== 'undefined') this.togglePopup(name);
+    };
+
+    const sorterFunction = (a: ISorterInput, b: ISorterInput) => {
+      if (a.type === 'undefined') {
+        return 0;
       }
-      return (
-        <DetailTableData
-          key={uid}
-          type={oneSpan.type}
-          name={oneSpan.name}
-          searchColor={searchColor}
-          values={values}
-          columnsArray={columnsArray}
-          color={color}
-          togglePopup={this.togglePopup}
-          valueNameSelector2={this.state.valueNameSelector2}
-          colorToPercent={colorToPercent}
-        />
-      );
-    });
-  }
+      if (b.type === 'undefined') {
+        return -1;
+      }
+      if (typeof a.index === 'number' && typeof b.index === 'number') return a.index - b.index;
+      return (a.index as string).localeCompare(b.index as string);
+    };
 
-  renderTableHead() {
-    const { sortAsc, sortIndex } = this.state;
-    return (
-      <tr>
-        {columnsArray.map((element: any, index: number) => (
-          <HeaderTable
-            element={element}
-            key={element.title}
-            sortIndex={sortIndex}
-            index={index}
-            sortClick={this.sortClick}
-            sortAsc={sortAsc}
-          />
-        ))}
-      </tr>
-    );
-  }
+    const onCellFunction = (record: ITableSpan) => {
+      if (this.props.uiFind && record.searchColor !== 'transparent') {
+        return {
+          style: { background: record.searchColor, borderColor: record.searchColor },
+        };
+      }
+      return {
+        style: { background: record.colorToPercent, borderColor: record.colorToPercent },
+      };
+    };
 
-  render() {
+    const columns: ColumnProps<ITableSpan>[] = [
```

Comment:
Because in each row of the table is ITableSpan type, as we already had a interface [ITableSpan](packages/jaeger-ui/src/components/TracePage/TraceStatistics/types.tsx) which has the required attributes to make table. So ITableSpan is used